lib/msf/core/modules/metadata/store.rb

```diff
@@ -124,4 +127,168 @@ def load_cache_from_file_store
     }
   end
 
+  # This method checks if the current module and library files match the cached checksum.
+  # It uses a per-file CRC32 cache to avoid recalculating checksums for files that haven't changed.
+  # If no cache exists, it will create one in the user's directory.
+  #
+  # @return [Boolean] True if the current checksum matches the cached one
+  def self.valid_checksum?
+    current_checksum = get_current_checksum
+    cached_sha = get_cached_checksum
+
+    # If no cached checksum exists, create the cache file with current checksum
+    if cached_sha.nil?
+      create_or_update_cache_file(get_cache_path, current_checksum)
+      return false
+    end
+
+    checksums_match?(current_checksum, cached_sha)
+  end
+
+  # Calculate the current checksum for all module and library files
+  # This calculates checksums for each file, caches them, and then
+  # generates an overall checksum from the individual file checksums.
+  #
+  # @return [String] The current overall checksum
+  def self.get_current_checksum
+    files = collect_files_to_check
+    per_file_cache_file = get_per_file_cache_path
+    per_file_cache = load_per_file_cache(per_file_cache_file)
+
+    file_crc32s_with_metadata = calculate_file_checksums(files, per_file_cache)
+
+    updated_cache = file_crc32s_with_metadata.to_h
```

Comment:
Per [RFC7159](https://datatracker.ietf.org/doc/html/rfc7159#section-1), JSON objects are unordered so to guarantee a deterministic calculation, I think this should be converted to an array and the file path which is currently the key, should be moved to a `path` key within the object. That would guarantee that the order is maintained where the objects are sorted by the file path.